avatars/avatars.js

```diff
@@ -53,1266 +31,92 @@ window.addEventListener('keydown', e => {
-      for (let i = 0; i < bones.length; i++) {
-        const bone = bones[i];
-        if (/hair/i.test(bone.name)) {
-          allHairBones.push(bone);
-        }
-        _recurseAllHairBones(bone.children);
-      }
-    };
-    _recurseAllHairBones(skeleton.bones);
-    const hairBones = tailBones.filter(bone => /hair/i.test(bone.name)).map(bone => {
-      for (; bone; bone = bone.parent) {
-        if (bone.parent === Head) {
-          return bone;
-        }
-      }
-      return null;
-    }).filter(bone => bone);
-    this.allHairBones = allHairBones;
-    this.hairBones = hairBones;
-
-    this.springBoneManager = null;
-    if (options.hair) {
-      const promiseHair = new Promise((resolve, reject) => {
-        if (!object) {
-          object = {};
-        }
-        if (!object.parser) {
-          object.parser = {
-            json: {
-              extensions: {},
-            },
-          };
-        }
-        if (!object.parser.json.extensions) {
-          object.parser.json.extensions = {};
-        }
-        if (!object.parser.json.extensions.VRM) {
-          object.parser.json.extensions.VRM = {
-            secondaryAnimation: {
-              boneGroups: this.hairBones.map(hairBone => {
-                const boneIndices = [];
-                const _recurse = bone => {
-                  boneIndices.push(this.allHairBones.indexOf(bone));
-                  if (bone.children.length > 0) {
-                    _recurse(bone.children[0]);
-                  }
-                };
-                _recurse(hairBone);
-                return {
-                  comment: hairBone.name,
-                  stiffiness: 0.5,
-                  gravityPower: 0.2,
-                  gravityDir: {
-                    x: 0,
-                    y: -1,
-                    z: 0,
-                  },
-                  dragForce: 0.3,
-                  center: -1,
-                  hitRadius: 0.02,
-                  bones: boneIndices,
-                  colliderGroups: [],
-                };
-              }),
-            },
-          };
-          object.parser.getDependency = async (type, nodeIndex) => {
-            if (type === 'node') {
-              return this.allHairBones[nodeIndex];
-            } else {
-              throw new Error('unsupported type');
-            }
-          };
-        }
-        resolve();
-      });
-      promiseHair.then();
-    }
+const loadData = new Promise((resolve, reject) =>{
+  let count = 2
 
-    // NeedsNewFunc: Pre-rotate
-    const preRotations = {};
-    const _ensurePrerotation = k => {
-      const boneName = modelBones[k].name;
-      if (!preRotations[boneName]) {
-        preRotations[boneName] = new THREE.Quaternion();
-      }
-      return preRotations[boneName];
-    };
-    if (flipY) {
-      _ensurePrerotation('Hips').premultiply(new THREE.Quaternion().setFromAxisAngle(new THREE.Vector3(1, 0, 0), -Math.PI / 2));
+  const progress = (err)=>{
```

Comment:
Can this be `Promise.all` rather than a custom callback counter?

Reply:
I try to make the code cleaner and more understandable, a lot of promises and async await make the code harder to understand, well, in general, we need a data loading system in the future